tests/test_address.py

```diff
@@ -1,144 +1,81 @@
-    )
-
-    assert address.id is not None
-    assert address.street1 == "118 2ND ST FL 4"
-    assert address.street2 == ""
-    assert address.country == "US"
+    assert isinstance(address, easypost.Address)
+    assert str.startswith(address.id, "adr_")
+    assert address.street1 == "388 TOWNSEND ST APT 20"
 
 
 @pytest.mark.vcr()
-def test_address_creation_with_verify_failure():
-    # Create an address with a verify parameter to test that it fails elegantly
-    address = easypost.Address.create(
-        verify=["delivery"],
-        street1="UNDELIEVRABLE ST",
-        city="San Francisco",
-        state="CA",
-        zip="94105",
-        country="US",
-        company="EasyPost",
-        phone="415-456-7890",
-    )
+def test_address_retrieve(basic_address):
+    address = easypost.Address.create(**basic_address)
+
+    retrieved_address = easypost.Address.retrieve(address.id)
 
-    assert address.id is not None
-    assert address.street1 == "UNDELIEVRABLE ST"
+    assert isinstance(retrieved_address, easypost.Address)
+    assert retrieved_address == address
 
-    assert address.verifications["delivery"]["success"] is False
 
-    assert len(address.verifications["delivery"]["errors"]) >= 2
-    assert address.verifications["delivery"]["errors"][0]["message"] == "Address not found"
-    assert address.verifications["delivery"]["errors"][1]["message"] == "House number is missing"
+@pytest.mark.vcr()
+def test_address_all(page_size):
+    addresses = easypost.Address.all(page_size=page_size)
+
+    addresses_array = addresses["addresses"]
+
+    assert len(addresses_array) <= page_size
+    assert addresses["has_more"] is not None
+    assert all(isinstance(address, easypost.Address) for address in addresses_array)
 
 
 @pytest.mark.vcr()
-def test_address_creation_with_verify_strict_failure():
-    # Create an address with a verify strict parameter to test that it fails elegantly
-    with pytest.raises(easypost.Error) as caught_exception:
-        easypost.Address.create(
-            verify_strict=["delivery"],
-            street1="UNDELIEVRABLE ST",
-            city="San Francisco",
-            state="CA",
-            zip="94105",
-            country="US",
-            company="EasyPost",
-            phone="415-456-7890",
-        )
-
-    exception = caught_exception.value.json_body
-
-    assert exception["error"]["code"] == "ADDRESS.VERIFY.FAILURE"
-    assert exception["error"]["message"] == "Unable to verify address."
-
-    assert len(exception["error"]["errors"]) >= 2
-    assert exception["error"]["errors"][0]["message"] == "Address not found"
-    assert exception["error"]["errors"][1]["message"] == "House number is missing"
+def test_address_create_verify(incorrect_address_to_verify):
+    """
+    Test creating a verified address.
+    We purposefully pass in slightly incorrect data to get the corrected address back once verified.
+    """
+    address = easypost.Address.create(**incorrect_address_to_verify)
```

Comment:
I will remove this function since the testing duplicate to the one you pointed out in the above comment: https://github.com/EasyPost/easypost-python/pull/162#discussion_r829439481